homeassistant/components/statistics/sensor.py

```diff
@@ -542,128 +543,130 @@ def _stat_average_step(self):
 
-    def _stat_datetime_newest(self):
+    def _stat_datetime_newest(self) -> StateType:
         if len(self.states) > 0:
-            return self.ages[-1]
+            return str(self.ages[-1])
         return None
 
-    def _stat_datetime_oldest(self):
+    def _stat_datetime_oldest(self) -> StateType:
         if len(self.states) > 0:
-            return self.ages[0]
+            return str(self.ages[0])
         return None
 
-    def _stat_distance_95_percent_of_values(self):
+    def _stat_distance_95_percent_of_values(self) -> StateType:
         if len(self.states) >= 2:
-            return 2 * 1.96 * self._stat_standard_deviation()
+            return 2 * 1.96 * cast(float, self._stat_standard_deviation())
         return None
 
-    def _stat_distance_99_percent_of_values(self):
+    def _stat_distance_99_percent_of_values(self) -> StateType:
         if len(self.states) >= 2:
-            return 2 * 2.58 * self._stat_standard_deviation()
+            return 2 * 2.58 * cast(float, self._stat_standard_deviation())
         return None
 
-    def _stat_distance_absolute(self):
+    def _stat_distance_absolute(self) -> StateType:
         if len(self.states) > 0:
             return max(self.states) - min(self.states)
         return None
 
-    def _stat_mean(self):
+    def _stat_mean(self) -> StateType:
         if len(self.states) > 0:
             return statistics.mean(self.states)
         return None
 
-    def _stat_median(self):
+    def _stat_median(self) -> StateType:
         if len(self.states) > 0:
             return statistics.median(self.states)
         return None
 
-    def _stat_noisiness(self):
+    def _stat_noisiness(self) -> StateType:
         if len(self.states) >= 2:
             diff_sum = sum(
                 abs(j - i) for i, j in zip(list(self.states), list(self.states)[1:])
             )
             return diff_sum / (len(self.states) - 1)
         return None
 
-    def _stat_quantiles(self):
+    def _stat_quantiles(self) -> StateType:
         if len(self.states) > self._quantile_intervals:
-            return [
-                round(quantile, self._precision)
-                for quantile in statistics.quantiles(
-                    self.states,
-                    n=self._quantile_intervals,
-                    method=self._quantile_method,
-                )
-            ]
+            return str(
+                [
+                    round(quantile, self._precision)
+                    for quantile in statistics.quantiles(
+                        self.states,
+                        n=self._quantile_intervals,
+                        method=self._quantile_method,
+                    )
+                ]
+            )
         return None
 
-    def _stat_standard_deviation(self):
+    def _stat_standard_deviation(self) -> StateType:
         if len(self.states) >= 2:
-            return statistics.stdev(self.states)
+            return float(statistics.stdev(self.states))
```

Comment:
Do we need to convert to `float` here, or should we cast?

Reply:
Good catch.